### PR TITLE
add extensions for i18next scanner

### PR DESCRIPTION
**Pull request by achen2401**
to include template html files in gil and eproms

### CODE REVIEW

File: portal/i18next_gulpfile.js [hunk 1]
```diff
@@ -50,21 +50,25 @@ function save(target) {
  */
 gulp.task('i18next-extraction', ['clean-src'], function() {
     console.log('extracting text and generate json file ...');
-    return gulp.src(['static/**/*.{js,html}'])
+    return gulp.src(['static/**/*.{js,html}', 'templates/*.html', 'gil/templates/gil/*.html', 'eproms/templates/eproms/*.html'])
```
Comment (ivan-c) [inline portal/i18next_gulpfile.js]: I just realized we'll need to separate ePROMs and TrueNTH content when uploading to Smartling- can you either add an option for specifying eproms/gil, or just split the extract command into two commands?

**Comment by achen2401** [comment]
@ivan-c  Ivan, I made additional changes here based on what we discussed.
running gulp command 'eproms', i.e.   gulp --gulpfile i18next_gulpfile.js eproms  will generate pot file containing eproms specific content
running gulp command 'truenth', i.e. gulp --gulpfile i18next_gulpfile.js truenth will generate pot file containing truenth specific content
the default gulp task will now generate pot file containing only extracted text from common files (e.g. main.js, templates/*.html)

Let me know if you see any issue with the changes.

File: portal/i18next_gulpfile.js [hunk 2]
```diff
@@ -46,70 +52,130 @@ function save(target) {
...
-    * converting json to pot
-    */
-   console.log('converting JSON to POT...');
-   return i18nextConv.i18nextToPot('en', fs.readFileSync(translationSourceDir+nameSpace+'.json'), options).then(save(srcPotFileName));
+    try {
+      console.log("source file, " + messageFilePath + ", found for locale: ", locale);
+      i18nextConv.gettextToI18next(locale, fs.readFileSync(messageFilePath), false)
+      .then(save(outputFileName));
+    } catch(e) {
+      console.log("Error occurred writing json from po file: ", messageFilePath);
```
[the remark this answers is not in the document]
Comment (achen2401) [reply portal/i18next_gulpfile.js]: Just a note here that console.log statement here will write output to terminal window when running the gulp task command.  Different from the console.log statement used to write to browser console.

File: portal/i18next_gulpfile.js [hunk 3]
```diff
@@ -8,33 +8,39 @@
...
+const translationDestinationDir = path.join(__dirname,"./static/files/locales/");
+
 /*
  * namespace
  */
-const nameSpace = 'frontend';
-const srcPotFileName = translationSourceDir+nameSpace+'.pot';
+const nameSpace = "frontend";
+const epromsNameSpace = "eproms";
+const truenthNameSpace = "truenth";
```
Comment (ivan-c) [inline portal/i18next_gulpfile.js]: Could you change this to "gil" to match Paul's directory names? Sorry for telling you "truenth" earlier!

File: portal/i18next_gulpfile.js [hunk 4]
```diff
@@ -120,100 +186,142 @@ gulp.task('combineAllPotFiles', ['i18nextConvertJSONToPOT'], function() {
...
 });
 
 /*
  * clean all generated destination json files
  */
-gulp.task('clean-dest', function() {
-  console.log('delete json files...')
-  return del([translationDestinationDir + '*/*.json']);
+gulp.task("clean-dest", function() {
+  console.log("delete json files...")
```
[the remark this answers is not in the document]
Comment (ivan-c) [reply portal/i18next_gulpfile.js]: Could you fix this? :)

File: portal/i18next_gulpfile.js [hunk 5]
```diff
@@ -120,100 +186,142 @@ gulp.task('combineAllPotFiles', ['i18nextConvertJSONToPOT'], function() {
...
+          });
+      });
+  });
+});
+
+/*
+ * clean eproms source file
+ */
+gulp.task("clean-eproms-src", function() {
+  console.log("delete source files...")
```
[the remark this answers is not in the document]
Comment (ivan-c) [reply portal/i18next_gulpfile.js]: Could you fix this? :)

**Comment by achen2401** [comment]
@ivan  Ivan, I replaced "truenth" with "gil".  The command to run to generate gil pot:
gulp --gulpfile i18next_gulpfile.js gil   Let me know if you see any issue with my fix.

**Pull request merged**